framework/bundles/org.jboss.tools.rsp.server/src/main/java/org/jboss/tools/rsp/server/secure/SecureStorageGuardian.java

```diff
@@ -0,0 +1,108 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Red Hat, Inc. Distributed under license by Red Hat, Inc.
+ * All rights reserved. This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution, and is
+ * available at http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors: Red Hat, Inc.
+ ******************************************************************************/
+package org.jboss.tools.rsp.server.secure;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import org.jboss.tools.rsp.api.ICapabilityKeys;
+import org.jboss.tools.rsp.api.RSPClient;
+import org.jboss.tools.rsp.api.dao.StringPrompt;
+import org.jboss.tools.rsp.secure.crypto.CryptoException;
+import org.jboss.tools.rsp.secure.model.ISecureStorage;
+import org.jboss.tools.rsp.secure.model.ISecureStorageProvider;
+import org.jboss.tools.rsp.secure.model.RSPSecureStorage;
+import org.jboss.tools.rsp.server.spi.client.ClientThreadLocal;
+import org.jboss.tools.rsp.server.spi.model.ICapabilityManagement;
+
+public class SecureStorageGuardian implements ISecureStorageProvider {
+	private File file;
+	private Map<RSPClient, byte[]> permissions;
+	private ISecureStorage storage = null;
+	public SecureStorageGuardian(File file) {
+		this.file = file;
+		this.permissions = new HashMap<>();
+	}
+	
+	public void addClient(RSPClient client, byte[] key) throws CryptoException {
+		RSPSecureStorage test = checkKey(key);
+		if( storage == null ) {
+			storage = test;
+		}
+		permissions.put(client, key);
+	}
+
+	public void removeClient(RSPClient client) {
+		permissions.remove(client);
+	}
+
+	private RSPSecureStorage checkKey(byte[] key) throws CryptoException {
+		RSPSecureStorage tmp = new RSPSecureStorage(file, key);
+		if( !file.exists()) {
+			tmp.load();
```

Comment:
So are you calling `load()` no matter what? Why not move that out of if/else and leave only `if' branch?